### ISSUE TITLE
y_ini - ftouch(filename)  Problem!

### ISSUE THREAD

**Issue by Walter-Correa** [pawn-lang/YSI-Includes#104]
Hello!
I use y_ini is 1 year and never had any problems.
Today I come home and access my VPS, see the server console and meeting:
_**YSI Error: INI_Open could not find or create file ...**_

No account was being created or opened and I was forced to restart the server!

Operating System: Windows Server 2003

**NOTE: This has never happened in one year! o.O**
Thank you for help me!

**Comment by Y-Less** [comment]
You're not giving me a lot to go on there.  One question though: why are you using a 13 year old OS?

**Comment by Walter-Correa** [comment]
I'm sorry for this! This has some influence on ftouch?
I use the same VPS (with the same OS) since 2014!

**Comment by Y-Less** [comment]
Well it shouldn't do, but I know some older versions of Windows had directory size limits.  If 2003 is one, and if that is the problem - I don't know, but you said everything was fine till this, so something must have changed.  I'm only thinking this because I had a similar problem in a totally unrelated project recently.

**Comment by Walter-Correa** [comment]
**I found this:**

•NTFS
NTFS, or "New Technology File System" introduced with Windows NT, is a completely redesigned file system.

◦Maximum disk size: 256 terabytes
◦Maximum file size: 256 terabytes
◦Maximum number of files on disk: 4,294,967,295
◦Maximum number of files in a single folder: 4,294,967,295

The disk of my VPS is NTFS!
**There would be another problem that is causing this?**

**Comment by Walter-Correa** [comment]
It happened again and I realized what happened to a single file and no other can be found after that. But every few seconds or minutes appeared the same problem for the same file and I concluded that the system was perhaps in a loop trying to find the file even exists because it was created when the player registered himself. There is no loop in my GM try to save the account of a player if it is not logged in and connected to the server. I believe that within the INI_Open function there is any problem that is causing the system to try to find the file, more than one time!

Maybe add a **return INI_NO_FILE;** inside the condition can stop this bug:
if (ftouch(filename) == -1)
{
        P:E("INI_Open could not find or create file %s", filename);
                **return INI_NO_FILE;**
}

No?

**Comment by Y-Less** [comment]
I think I get what you're trying to say, but I'm not sure sorry.  Can you try clarify it a bit please?

**Comment by Walter-Correa** [comment]
Okay, come on ... When checking the console and server_logs.txt realized that this error happened with only 1 file related to a user account. Checked the folder scriptfiles/ContasID/3270.ini and the file was there. I concluded that the system was unable to read the file, because it exists and then had been created correctly. Only appeared in the logs the problem with this file specifically and no other account was created or read after that, all players complaining that your account has been reset, but in fact was not being read only. It all happened after the error with the above mentioned file. I realized that every seconds or minutes the same mistake with the same file appeared, but the player was not more connected. I believe there is some loop in your system which makes him trying to read or create the file more than one time because my saving system only works when the player is connected and logged(PlayerLogged [playerid] == 1) and do not work with loops to save accounts , but each player has a PVarInt that makes the count individually and save the account every 5 minutes after disconnecting the PVarInt is reseted. What do you think is happening? As the ftouch work? Thanks man, I love your jobs and creations!

**Comment by Walter-Correa** [comment]
Y_Less!
I discovered the reason: There is a bug in the Windows that appear to date has not been fixed. After several files created / moved / deleted the system does not refresh the folders with the current files causing this bug until you press F5. This is the reason that is generating the error cited above.

I found a solution by sorting the files in the folders by Name.

Thank you for your attention and say that is not a mistake of your include y_ini!
